src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -147,50 +164,129 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
+                    return await ExtractInterfaceToNewFileAsync(
+                        solution: solution,
+                        containingNamespaceDisplay: containingNamespaceDisplay,
+                        extractedInterfaceSymbol: extractedInterfaceSymbol,
+                        refactoringResult: refactoringResult,
+                        extractInterfaceOptions: extractInterfaceOptions,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                case ExtractInterfaceOptionsResult.ExtractLocation.SameFile:
+                    return await ExtractInterfaceToSameFileAsync(
+                        solution: solution,
+                        extractedInterfaceSymbol: extractedInterfaceSymbol,
+                        refactoringResult: refactoringResult,
+                        extractInterfaceOptions: extractInterfaceOptions,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                default: throw new InvalidOperationException($"Unable to extract interface for operation of type {extractInterfaceOptions.GetType()}");
+            }
+        }
 
-            var unformattedInterfaceDocument = GetUnformattedInterfaceDocument(
+        private async Task<ExtractInterfaceResult> ExtractInterfaceToNewFileAsync(
+            Solution solution, string containingNamespaceDisplay, INamedTypeSymbol extractedInterfaceSymbol,
+            ExtractInterfaceTypeAnalysisResult refactoringResult, ExtractInterfaceOptionsResult extractInterfaceOptions, CancellationToken cancellationToken)
+        {
+            var symbolMapping = await CreateSymbolToDeclarationAnnotationMapAsync(
+                extractInterfaceOptions.IncludedMembers,
                 solution,
+                refactoringResult.TypeNode,
+                cancellationToken).ConfigureAwait(false);
+
+
+
+            var interfaceDocumentId = DocumentId.CreateNewId(refactoringResult.DocumentToExtractFrom.Project.Id, debugName: extractInterfaceOptions.FileName);
+
+            var unformattedInterfaceDocument = await GetUnformattedInterfaceDocumentAsync(
+                symbolMapping.AnnotatedSolution,
                 containingNamespaceDisplay,
                 extractInterfaceOptions.FileName,
                 refactoringResult.DocumentToExtractFrom.Folders,
                 extractedInterfaceSymbol,
                 interfaceDocumentId,
-                cancellationToken);
+                cancellationToken).ConfigureAwait(false);
 
-            var solutionWithFormattedInterfaceDocument = GetSolutionWithFormattedInterfaceDocument(unformattedInterfaceDocument, cancellationToken);
+            var solutionWithFormattedInterfaceDocument = await GetFormattedSolutionAsync(
+                unformattedInterfaceDocument.Project.Solution,
+                SpecializedCollections.SingletonList(unformattedInterfaceDocument.Id),
+                cancellationToken).ConfigureAwait(false);
```

Comment:
this call looks strange to me.  Based on the signature... it looks like you intend to only format the unformattedInterfaceDocument.  But, in that case, i would expect to just format that document and not do anything with the solution.  I need to look further to see what's going onhere.

Reply:
Good point, `GetSolutionWithOriginalTypeUpdatedAsync` performs a formatting. Should probably remove the formatting from that call and format all at once in this method.